interp/interp.go

```diff
@@ -207,16 +212,32 @@ func (n *node) Walk(in func(n *node) bool, out func(n *node)) {
 
 // Options are the interpreter options.
 type Options struct {
-	// GoPath sets GOPATH for the interpreter
+	// GoPath sets GOPATH for the interpreter.
 	GoPath string
-	// BuildTags sets build constraints for the interpreter
+	// BuildTags sets build constraints for the interpreter.
 	BuildTags []string
+	// Standard input, output and error streams.
+	Stdin          io.Reader
+	Stdout, Stderr io.Writer
 }
 
 // New returns a new interpreter.
 func New(options Options) *Interpreter {
+	in := options.Stdin
+	if in == nil {
+		in = os.Stdin
+	}
+	out := options.Stdout
+	if out == nil {
+		out = os.Stdout
+	}
+	err := options.Stderr
+	if err == nil {
+		err = os.Stderr
+	}
```

Comment:
it's a little bit inconsistent that all of this is done here before creating i, instead of being done after, with the rest of the options, i.e. L252, isn't it?

Reply:
ok.